build/build_cpp.sh

```diff
@@ -63,12 +76,24 @@ else
 	else
 		echo "Usage: ./build_cpp.sh 	[-e|--example] [example name]|all"
 		echo "			[-b|--benchmark] [benchmark name]|all"
+		case `uname` in 
+			Darwin*) echo "			Xcode" ;;
+			**) ;;
+		esac
 		exit -1
 	fi
 fi
 
-# Make
-make
+# Make / Build
+if [ "$1" == "Xcode" ]; then
+	case `uname` in 
+		Darwin*) xcodebuild && open ./;;
+		**) echo "Xcode Project generated at ../llama_xcode/llama_cpp.xcodeproj" ;;
+	esac
+else
+	make
+fi
+
 if [ $? != 0 ]; then
```

Comment:
Move this "if" into above else case. And write conditions for Xcode generation failed.